objfile/scanner.go

```diff
@@ -94,111 +69,103 @@ var PPC_BE_sig = signatureModuleDataInitPPC{2, 6, `rule PPC_BEfirstmoduledata
 // 0x000000000005C1F4 21 18 41 F9        LDR             X1, [X1,#0x230]
 // 0x000000000005C1F8 21 0D 00 B4        CBZ             X1, loc_5C39C   0xb4000d21
 // THIS SIG ENCODES the 0x230 struct field offset - might need to mask that more if we see misses - TODO
-var ARM64_sig = signatureModuleDataInitARM64{0, 4, `rule ARM64firstmoduledata
-{
-    strings:
-        $sig = { ?? ?? ?? (90 | b0 | f0 | d0) ?? ?? ?? 91 ?? ?? ?? (14 | 17) ?? ?? 41 F9 ?? ?? ?? B4 }
-    condition:
-        $sig
-}`, "ARM64"}
+var ARM64_sig = signatureModuleDataInitARM64{0, 4, `{ ?? ?? ?? (90 | b0 | f0 | d0) ?? ?? ?? 91 ?? ?? ?? (14 | 17) ?? ?? 41 F9 ?? ?? ?? B4 }`}
 
 // 0x0006AA00 80 12 9F E5    LDR             R1, =firstmoduleData   // 0xE59F1280 -> 0b11 100101100111110001001010000000 -> size = 11,
 // 0x0006AA04 00 00 00 EA    B               loc_6AA0C
 // 0x0006AA08 18 11 91 E5    LDR             R1, [R1,#0x118]
 // 0x0006AA0C 00 00 51 E3    CMP             R1, #0
 // 0x0006AA10 69 00 00 0A    BEQ             loc_6ABBC
-var ARM32_sig = signatureModuleDataInitARM32{0, `rule ARM32firstmoduledata
-{
-    strings:
-        $sig = { ?? ?? 9F E5 ?? ?? ?? EA ?? ?? ?? E5 ?? ?? ?? E3 ?? ?? ?? 0A }
-    condition:
-        $sig
-}`, "ARM32"}
+var ARM32_sig = signatureModuleDataInitARM32{0, `{ ?? ?? 9F E5 ?? ?? ?? EA ?? ?? ?? E5 ?? ?? ?? E3 ?? ?? ?? 0A }`}
 
 func findModuleInitPCHeader(data []byte, sectionBase uint64) []SignatureMatch {
 	var matches []SignatureMatch = make([]SignatureMatch, 0)
 
-	c, _ := yara.NewCompiler()
-	c.AddString(x64sig.signature, x64sig.namespace)
-	c.AddString(x86sig.signature, x86sig.namespace)
-	c.AddString(PPC_BE_sig.signature, PPC_BE_sig.namespace)
-	c.AddString(ARM64_sig.signature, ARM64_sig.namespace)
-	c.AddString(ARM32_sig.signature, ARM32_sig.namespace)
-	rules, err := c.GetRules()
-	if err != nil {
-		return matches
+	g, e := NewBinaryRegexpGroup(map[string]string{
+		"x64":    x64sig.signature,
+		"x86":    x86sig.signature,
+		"ARM64":  ARM64_sig.signature,
+		"ARM32":  ARM32_sig.signature,
+		"PPC_BE": PPC_BE_sig.signature,
+	})
+	if e != nil {
+		// programmer error: check the patterns
+		panic(e)
```

Comment:
I don't love panicking here, I understand it's a 'static' error due to hardcoded patterns but this would break external systems that try to parse JSON output. Maybe this is a +1 for hard-coding regex patterns instead - we'd remove the error case entirely? If only Go had compile-time logic.

As this is basically an assert() we can keep it but just noting it.